x/ibc/02-client/types/tendermint/evidence.go

```diff
@@ -51,41 +50,54 @@ func (ev Evidence) Hash() cmn.HexBytes {
 
 // ValidateBasic implements Evidence interface
 func (ev Evidence) ValidateBasic() error {
-	if ev.DuplicateVoteEvidence == nil {
-		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "duplicate evidence is nil")
+	// ValidateBasic on both validators
+	if err := ev.Header1.ValidateBasic(ev.ChainID); err != nil {
+		return errors.ErrInvalidEvidence(errors.DefaultCodespace, err.Error())
 	}
-	err := ev.DuplicateVoteEvidence.ValidateBasic()
-	if err != nil {
+	if err := ev.Header2.ValidateBasic(ev.ChainID); err != nil {
 		return errors.ErrInvalidEvidence(errors.DefaultCodespace, err.Error())
 	}
-	if ev.ChainID == "" {
-		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "chainID is empty")
+	if ev.Header1.ValidatorSet == nil || ev.Header2.ValidatorSet == nil {
+		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "validator set in header is nil")
 	}
-	if ev.ValidatorPower <= 0 {
-		return errors.ErrInvalidEvidence(errors.DefaultCodespace, fmt.Sprintf("invalid Validator Power: %d", ev.ValidatorPower))
+	// Ensure that Heights are the same
+	if ev.Header1.Height != ev.Header2.Height {
+		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "headers in evidence are on different heights")
 	}
-	if ev.TotalPower < ev.ValidatorPower {
-		return errors.ErrInvalidEvidence(errors.DefaultCodespace, fmt.Sprintf("invalid Total Power: %d", ev.TotalPower))
+	// Ensure that Commit Hashes are different
+	if ev.Header1.Commit.BlockID.Equals(ev.Header2.Commit.BlockID) {
+		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "Headers commit to same blockID")
 	}
-	return nil
-}
 
-// GetConsensusAddress implements exported.Evidence interface
-func (ev Evidence) GetConsensusAddress() sdk.ConsAddress {
-	return sdk.ConsAddress(ev.DuplicateVoteEvidence.Address())
-}
+	// Ensure that validator sets that voted on differing headers are the same validators
```

Comment:
Why would we need to check this? They might be different validators (e.g. if bisection happened).